tests/components/smartthings/test_config_flow.py

```diff
@@ -224,128 +283,159 @@ async def test_cloudhook_app_created_then_show_wait_form(
-    assert result["step_id"] == "wait_install"
-    assert result["errors"] == {"base": "app_not_installed"}
+    assert result["step_id"] == "select_location"
+    assert result["data_schema"]({CONF_LOCATION_ID: location.location_id}) == {
+        CONF_LOCATION_ID: location.location_id
+    }
 
 
-async def test_config_entry_created_when_installed(
-    hass, location, installed_app, smartthings_mock
-):
-    """Test a config entry is created once the app is installed."""
+async def test_step_select_location_aborts(hass, location, smartthings_mock):
+    """Test select location aborts if no available locations."""
+    smartthings_mock.locations.return_value = [location]
+    entry = MockConfigEntry(
+        domain=DOMAIN, data={CONF_LOCATION_ID: location.location_id}
+    )
+    entry.add_to_hass(hass)
     flow = SmartThingsFlowHandler()
     flow.hass = hass
-    flow.access_token = str(uuid4())
-    flow.app_id = installed_app.app_id
     flow.api = smartthings_mock
-    flow.oauth_client_id = str(uuid4())
-    flow.oauth_client_secret = str(uuid4())
-    data = {
-        CONF_REFRESH_TOKEN: str(uuid4()),
-        CONF_LOCATION_ID: installed_app.location_id,
-        CONF_INSTALLED_APP_ID: installed_app.installed_app_id,
-    }
-    hass.data[DOMAIN][CONF_INSTALLED_APPS].append(data)
 
-    result = await flow.async_step_wait_install({})
+    result = await flow.async_step_select_location()
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "no_available_locations"
 
-    assert not hass.data[DOMAIN][CONF_INSTALLED_APPS]
-    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-    assert result["data"]["app_id"] == installed_app.app_id
-    assert result["data"]["installed_app_id"] == installed_app.installed_app_id
-    assert result["data"]["location_id"] == installed_app.location_id
-    assert result["data"]["access_token"] == flow.access_token
-    assert result["data"]["refresh_token"] == data[CONF_REFRESH_TOKEN]
-    assert result["data"]["client_secret"] == flow.oauth_client_secret
-    assert result["data"]["client_id"] == flow.oauth_client_id
-    assert result["title"] == location.name
 
+async def test_step_select_location_advances(hass):
+    """Test select location aborts if no available locations."""
+    location_id = str(uuid4())
+    app_id = str(uuid4())
+    flow = SmartThingsFlowHandler()
+    flow.hass = hass
+    flow.app_id = app_id
+
+    result = await flow.async_step_select_location({CONF_LOCATION_ID: location_id})
 
-async def test_multiple_config_entry_created_when_installed(
-    hass, app, locations, installed_apps, smartthings_mock
-):
-    """Test a config entries are created for multiple installs."""
-    assert await async_setup_component(hass, "persistent_notification", {})
+    assert flow.location_id == location_id
+    assert result["type"] == data_entry_flow.RESULT_TYPE_EXTERNAL_STEP
+    assert result["step_id"] == "authorize"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5359319)